src/event_imp.rs

```diff
@@ -1044,6 +1044,216 @@ fn test_debug_ready() {
+    #[inline]
+    pub fn both() -> Interests {
+        Interests(READABLE | WRITABLE)
+    }
+    
+    /// Returns true if the value includes readable readiness
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::readable();
+    ///
+    /// assert!(interest.is_readable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    pub fn is_readable(&self) -> bool {
+        (self.0 & READABLE) != 0
+    }
+
+    /// Returns true if the value includes writable readiness
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::writable();
+    ///
+    /// assert!(interest.is_writable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    pub fn is_writable(&self) -> bool {
+        (self.0 & WRITABLE) != 0
+    }
+
+    /// Create a `Interests` instance using the given `usize` representation.
+    ///
+    /// The `usize` representation must be treated as opaque. There is no
+    /// guaranteed correlation between the returned value and platform defined
+    /// constants. Also, there is no guarantee that the `usize` representation
+    /// will remain constant across patch releases of Mio.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::readable();
+    /// let interest_usize = interest.as_usize();
+    /// let interest2 = Interests::from_usize(interest_usize);
+    ///
+    /// assert_eq!(interest, interest2);
+    /// ```
+    pub fn from_usize(val: usize) -> Interests {
+        Interests(val)
+    }
+    
+    /// Returns a `usize` representation of the `Interests` value.
+    ///
+    /// This `usize` representation must be treated as opaque. There is no
+    /// guaranteed correlation between the returned value and platform defined
+    /// constants. Also, there is no guarantee that the `usize` representation
+    /// will remain constant across patch releases of Mio.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::readable();
+    /// let interest_usize = interest.as_usize();
+    /// let interest2 = Interests::from_usize(interest_usize);
+    ///
+    /// assert_eq!(interest, interest2);
+    /// ```
+    pub fn as_usize(&self) -> usize {
+        self.0
+    }
+}
+
+impl<T: Into<Interests>> ops::BitOr<T> for Interests {
+    type Output = Interests;
+
+    #[inline]
+    fn bitor(self, other: T) -> Interests {
+        Interests(self.0 | other.into().0)
+    }
+}
+
+impl<T: Into<Interests>> ops::BitOrAssign<T> for Interests {
+    #[inline]
+    fn bitor_assign(&mut self, other: T){
+        self.0 |= other.into().0;
+    }
+}
+
+impl<T: Into<Interests>> ops::Sub<T> for Interests {
```

Comment:
Do we want to commit to this? Are there a lot of cases where you want to remove interests?